drakrun/drakrun/lib/postprocessing/capa_plugin/capa_processor.py

```diff
@@ -0,0 +1,428 @@
+        # we use the injected processes log to get the malware process' parent pid
+        injected_processes = [orjson.loads(line) for line in f]
+
+    with pstree_path.open("r") as f:
+        # we use the process tree to get all of the malware process' child processes
+        pstree = orjson.loads(f.read())
+
+    # make sure we have the right inject.log entry (using the metadata.json file)
+    malware_injection_log: str = next(
+        filter(
+            lambda line: line["ProcessName"] == metadata["start_command"],
+            injected_processes,
+        )
+    )
+
+    # get the parent malware process' pid and process name
+    malware_pid: int = malware_injection_log["InjectedPid"]
+    malware_procname: str = malware_injection_log["ProcessName"]
+
+    # find the malware process' sub-tree in the overall process tree
+    try:
+        malware_process: Dict = find_process_in_pstree(
+            pstree, malware_pid, malware_procname
+        )
+    except ValueError:
+        # malware pid not found, analyze entire log instead
+        return None
+
+    # get all child processes of the malware process
+    malware_pids = get_all_child_processes(malware_process)
+
+    # return a list of all spawned malware processes (does not include process injection)
+    return malware_pids
+
+
+def get_drakvuf_feature_extractor(
+    calls: Iterator[Dict],
+) -> capa.features.extractors.drakvuf.extractor.DrakvufExtractor:
+    # wrapper routing for initializing the Drakvuf feature extractor
+    return capa.features.extractors.drakvuf.extractor.DrakvufExtractor.from_report(
+        calls
+    )
+
+
+def decode_json_lines(fd: BinaryIO) -> Iterator[Dict]:
+    # read and decode json lines into a list of dictionaries
+    for line in fd:
+        try:
+            # we use orjson for a small performance improvement
+            line_s = line.strip()
+            obj = orjson.loads(line_s)
+            yield obj
+        except (orjson.JSONDecodeError):
+            # sometimes Drakvuf reports bad method names and/or malformed JSON
+            logger.debug("bad drakvuf log line: %s", line)
+
+
+def dynamic_capa_analysis(
+    analysis_dir: Path,
+    rules: capa.rules.RuleSet,
+    malware_pids: Optional[List[int]] = None,
+) -> Tuple[Path, capa.render.result_document.MatchResults]:
+
+    # save all api calls and native calls into one list which gets sorted by capa later on.
+    calls = []
+
+    # read api calls
+    if (analysis_dir / "apimon.log").exists():
+        with (analysis_dir / "apimon.log").open("r", errors="ignore") as apimon_fd:
+            calls += list(decode_json_lines(apimon_fd))
+    else:
+        logger.debug("missing apimon.log file")
+
+    # read syscalls
+    if (analysis_dir / "syscall.log").exists():
+        with (analysis_dir / "syscall.log").open("r", errors="ignore") as syscall_fd:
+            calls += list(decode_json_lines(syscall_fd))
+    else:
+        logger.debug("missing syscall.log file")
+
```

Comment:
nice to have - maybe raise an exception `if not calls` (so there's neither syscall.log nor apimon.log)